qiskit_nature/problems/sampling/protein_folding/bead_contacts/contact_map.py

```diff
@@ -0,0 +1,124 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""A class that stores contacts between beads of a peptide as qubit operators."""
+from typing import DefaultDict, List, Union
+
+from qiskit.opflow import PauliSumOp, PauliOp
+
+from ..bead_contacts.contact_map_builder import (
+    _create_contact_qubits,
+)
+from ..peptide.pauli_ops_builder import _build_full_identity
+from ..peptide.peptide import Peptide
+
+
+class ContactMap:
+    """A class that stores contacts between beads of a peptide as qubit operators."""
+
+    def __init__(self, peptide: Peptide):
+        self._peptide = peptide
+        (
+            self._lower_main_upper_main,
+            self._lower_side_upper_main,
+            self._lower_main_upper_side,
+            self._lower_side_upper_side,
+            self.num_contacts,
+        ) = _create_contact_qubits(peptide)
+
+        """
+
+       Args:
+           peptide: A Peptide object that includes all information about a protein.
+       """
+
+    @property
+    def peptide(self) -> Peptide:
+        """Returns a peptide."""
+        return self._peptide
+
+    @property
+    def lower_main_upper_main(self) -> DefaultDict[int, dict]:
```

Comment:
Do you need to return `DefaultDict`? you can use `dict()` to convert it to dict?

Reply:
I could, by do you know how expensive is the conversion? The data structure is used internally so likely the type does not make much difference.

Reply:
You can probably even use `Dict[int, dict]` as the return-type. Maybe you can also detail `dict` to a finer degree